app/workers/activitypub/fetch_all_replies_worker.rb

```diff
@@ -49,20 +49,39 @@ def perform(root_status_id, options = {})
 
   private
 
-  def get_replies(status_uri, max_pages, options = {})
-    replies_collection_or_uri = get_replies_uri(status_uri)
+  # @param status [String, Hash]
+  #   status URI, or the prefetched body of the Note object
+  def get_replies(status, max_pages, options = {})
+    replies_collection_or_uri = get_replies_uri(status)
     return if replies_collection_or_uri.nil?
 
-    ActivityPub::FetchAllRepliesService.new.call(status_uri, replies_collection_or_uri, max_pages: max_pages, **options.deep_symbolize_keys)
+    ActivityPub::FetchAllRepliesService.new.call(value_or_id(status), replies_collection_or_uri, max_pages: max_pages, **options.deep_symbolize_keys)
   end
 
-  def get_replies_uri(parent_status_uri)
-    fetch_resource(parent_status_uri, true)&.fetch('replies', nil)
+  # Get the URI of the replies collection of a status
+  #
+  # @param parent_status [String, Hash]
+  #   status URI, or the prefetched body of the Note object
+  def get_replies_uri(parent_status)
+    resource = parent_status.is_a?(Hash) ? parent_status : fetch_resource(parent_status, true)
+    resource&.fetch('replies', nil)
   rescue => e
-    Rails.logger.info { "FetchAllRepliesWorker - #{@root_status.uri}: Caught exception while resolving replies URI #{parent_status_uri}: #{e} - #{e.message}" }
+    Rails.logger.info { "FetchAllRepliesWorker - #{@root_status.uri}: Caught exception while resolving replies URI #{parent_status}: #{e} - #{e.message}" }
     # Raise if we can't get the collection for top-level status to trigger retry
-    raise e if parent_status_uri == @root_status.uri
+    raise e if value_or_id(parent_status) == @root_status.uri
 
     nil
   end
+
+  # Get the root status, updating the status without fetching it twice
+  #
+  # @param root_status_uri [String]
+  def get_root_replies(root_status_uri, options = {})
+    root_status_body = fetch_resource(root_status_uri, true)
+    raise RuntimeError("FetchAllRepliesWorker - #{@root_status.uri}: Root status could not be fetched") if root_status_body.nil?
+
+    FetchReplyWorker.perform_async(root_status_uri, { prefetched_body: root_status_body })
```

Comment:
I would still pass `options` there as we want to pass at least `request_id`.

Reply:
done in https://github.com/mastodon/mastodon/pull/34615/commits/aeccd8d07d9cdaeed3de5f0d938dbd842cd33649